kernel/mempool.c

```diff
@@ -172,14 +172,25 @@ void k_free(void *ptr)
 
 #if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
 
+#if (CONFIG_HEAP_MEM_POOL_SIZE < CONFIG_HEAP_MEM_POOL_MIN_SIZE)
+#error "The smallest object must have a lower size then the entire heap"
```

Comment:
Seems like if you want this to be a build error, you should just put it in the mempool declaration macro.  There's nothing specific to the system heap that has this requirement.